### DISCUSSION TITLE
convert stdpopsim models to msprime demography model

### DISCUSSION THREAD

**Discussion by lakishadavid** [tskit-dev/msprime#2148] [Q&A] [answered]
How do I convert the stdpopsim model to a msprime.Demography instance?

I see the example of the demography model in python, but how would I use the managed stdpopsim models in msprime? I can't use stdpopsim directly because I need chromosome-aware simulations, and as far as I can tell, stdpopsim does not model that. I can successfully build the demographic model and simulate data as per the tutorials would rather use the managed models in stdpopsim. 

Also, I don't see [Africa_1B08](https://popsim-consortium.github.io/stdpopsim-docs/stable/catalog.html#sec_catalog_homsap_models_africa_1b08) in `species.demographic_models` even though it's in the documentation. Is this one no longer being maintained? (I'm using stdpopsim-0.2.0)

Here is my code:
```
import stdpopsim

species = stdpopsim.get_species("HomSap")
model = species.get_demographic_model("AmericanAdmixture_4B11")
    
completed_ts = msprime.sim_ancestry(
    recombination_rate = rate_map,
    demography = model,
    model=[
        msprime.DiscreteTimeWrightFisher(duration=20),
        msprime.StandardCoalescent(),
    ],
    discrete_genome=True,
    samples={
        "AFR" : 300, 
        "ADMIX" : 300},
    record_provenance=False,
    random_seed = seed + 200
    )
```

and this is the error
`TypeError: demography argument must be an instance of msprime.Demography`

**Comment by nspope** [comment, marked as answer]
Hi, the `msprime.Demography` object is stored as an attribute called `model` in the `stdpopsim.DemographicModel` object. So, something like:
```python
import stdpopsim

species = stdpopsim.get_species("HomSap")
demog_model = species.get_demographic_model("AmericanAdmixture_4B11")
    
completed_ts = msprime.sim_ancestry(
    recombination_rate = rate_map,
    demography = demog_model.model, #<<--- here
    ...
```

As for `HomSap/Africa_1B08`, it should still be in there! The following works for me with stdpopsim 0.2.0:
```python
import stdpopsim
homsap = stdpopsim.get_species("HomSap")
demog_model = homsap.get_demographic_model("Africa_1B08")
```
